feat(gh-4327): add tag ids to export

sayedMurtadha opened the pull request:
closes #4327

dtslvr commented:
Hey @sayedMurtadha

Please leave a comment in https://github.com/ghostfolio/ghostfolio/issues/4327 that I can assign you 🙂

dtslvr commented on apps/api/src/app/export/export.service.ts:
```diff
@@ -92,7 +93,8 @@ export class ExportService {
                 ? SymbolProfile.name
-                : SymbolProfile.symbol
+                : SymbolProfile.symbol,
+            tags: tags.map(({ id: tagId }) => tagId)
```
Could you use the explicit form for consistency please?
```
{ return tagId; }
```

sayedMurtadha replied:
done

dtslvr commented on apps/api/src/app/export/export.service.ts:
```diff
@@ -73,7 +73,8 @@ export class ExportService {
           type,
-          unitPrice
+          unitPrice,
+          tags
```
Move up to keep the alphabetical order ⬆️

dtslvr reviewed (changes requested):
Thanks for your contribution, @sayedMurtadha 😎

I have reviewed your changes and added a few comments. Could you also include all used tags (cf. `accounts`) as stated in the description?

dtslvr commented on apps/api/src/app/export/export.service.ts:
```diff
@@ -43,6 +45,19 @@ export class ExportService {
       }
     );
+    const allTags = (
+      await this.tagService.getTags({
```
You can use `this.tagService.getTagsForUser(userId)` and filter the result by `isUsed`

dtslvr reviewed (changes requested):
I have reviewed your changes again and added a few comments.

By the way, I'm not able to start the server anymore. You need to import `TagModule` in `ExportModule` to have access to `TagService`.

sayedMurtadha commented:
@dtslvr I amended the changes and rebased my branch on the latest main commit, I updated `CHANGELOG.md` as well, thanks for your patience reviewing this PR as it is my first time using `NestJs`.

dtslvr reviewed (approved):
Thanks for your contribution, @sayedMurtadha! 🙌🏻
I have slightly refactored the code. Ready to merge now.
